persistence.js

```diff
@@ -598,19 +564,22 @@ RedisPersistence.prototype.streamWill = function (brokers) {
   return stream
 }
 
+// Only used from tests
 RedisPersistence.prototype.getClientList = function (topic) {
-  var entries = this._trie.match(topic, topic)
-
-  function pushClientList (size, next) {
-    if (entries.length === 0) {
-      return next(null, null)
+  const subKey = subscribersKey + topic
+  var stream = new Stream()
+  this._db.hgetall(subKey, function clientHash (err, data) {
+    if (err) {
+      stream.emit(err)
+      return
     }
-    var chunk = entries.slice(0, 1)
-    entries = entries.slice(1)
-    next(null, chunk[0].clientId)
-  }
-
-  return from.obj(pushClientList)
+    var entries = Object.keys(data)
+    for (const entry of entries) {
+      stream.emit('data', entry)
+    }
+    stream.emit('end')
+  })
+  return stream
 }
```

Comment:
I would try to implement this similar to how it was implemented before using `from2` library - it's designed to handle backpressure properly. But the problem is that this method would have to become async, or you would have to instantiate the stream first anyway and then probably you would have to populate the existing stream anyway 🤔 

ok, since this method is for tests only, then let's stick with whatever works - I tried to figure out what would be the best way to create a clean readable stream here while performing an async operation and somehow keeping this method sync, and I couldn't find anything straightforward, so let's just move on :sweat_smile:

Reply:
😄 👍